lib/bigtable/filter.js

```diff
@@ -0,0 +1,248 @@
+/*!
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /*!
+ * @module bigtable/filter
+ */
+
+'use strict';
+
+var is = require('is');
+var Entry = require('./entry.js');
+
+/**
+ *
+ */
```

Comment:
@stephenplusplus so I think way back when we initially talked about Filters, we talked about doing a plain old object (which we could probably still achieve), however because of the complexity of filters I ended up making a class instead. Also this mirrors Datastore Queries.

```js
var filter = table.createFilter()
  .family('name')
  .column('first');
```

Where it gets more complicated is using interleaving or conditional filters.

```js
// this is a weak example of an interleaving filter, but you get the idea
var filter = table.createFilter()
  .interleave([
    table.createFilter().family('name').column('first'),
    table.createFilter().family('life').column('born')
  ])
  .cellOffset(5)
  .cellLimit(10);
```

We could still go with an object if you feel that would be better, it would looks something like..

```js
var filter = {
  interleave: [{
    family: 'name',
    column: 'first'
  }, {
    family: 'life',
    column: 'born'
  }],
  cellOffset: 5,
  cellLimit: 10
};
```